### DISCUSSION TITLE
Livewire and Summernote text editor

### DISCUSSION THREAD

**Discussion by SigalZ** [livewire/livewire#4523] [1. Help]
Hello,

I'm new to Livewire and have a problem with Summernote text editor.

I have a form component where I do live validations like this:
`
public function updated($name)
{
     $this->validateOnly($name);
}
`

If I fill in an input field on the form, it goes to the component's validation comes back, summernote disappears and I'm left with a plain text-area.

I can't wrap it with a wire:ignore div because then it is not being validated.

I tried adding this in the blade scripts:

`
Livewire.on('onValidation', () => {
        $('.summernote').summernote();
    });
`

and in the component:

`
public function updated($name)
{
        $this->emit('onValidation');
        $this->validateOnly($name);        
}
`

or this:

`
public function updated($name)
{
     $this->validateOnly($name);        
      $this->emit('onValidation');
}
`

Nothing works.

Also, if I refresh the page, the values in the text-editors stay and do not clear.

Can someone please help?

**Comment by joshhanley** [comment]
@SigalZ as Summernote is a Javascript library that modifies the DOM, you will need to use `wire:ignore` otherwise Livewire wipes it out, hence why you are seeing it disappear. See here in the docs https://laravel-livewire.com/docs/2.x/alpine-js#ignoring-dom-changes

But that shouldn't impact validation, but you need just make sure your validation error message is outside of the element you put the `wire:ignore` on.

Hope this helps!

**Reply by SigalZ** [reply]
Thank you @joshhanley but I still have a problem that maybe you can help with?

If I delete the text in the textarea, I get the error, but when I refresh the page, I don't get the value form the database again.
The textarea stays empty.

My field looks like this now:
```
<div class="form-group" wire:ignore>
     <label class="control-label" for="shortDesc">Short Desctiption</label>
     <textarea wire:model="shortDesc" class="form-control" name="short_desc" id="shortDesc">{!! $blend->short_desc !!}</textarea>
</div>
 @error('shortDesc')
         <div role="error" style="font-size: 80%; color: #dc3545" class="mb-4">
             <strong>{{ $message }}</strong>
         </div>
 @enderror
```
On the component:
```
public function mount(Coffee $blend)
{
        $this->blend = $blend;
        
        if(!is_null($this->blend->id)) {
            $this->shortDesc = $this->blend->short_desc;
        }
}
```

I'm not sure if I need to do bind it to the livewire (wire:model="shortDesc") because it doesn't get the value from it anyway. I have to put {!! $blend->short_desc !}} to get the value.

and on page refresh it doesn't display again.

Just a note if anyone is interested:
I found out the error didn't display because of the invalid-feedback class. 
I moved it out of the wire:ignore div and it still didn't display but when I removed the invalid-feedback class, it started working.

**Reply by SigalZ** [reply]
Btw, in my scripts I have this code:
```
$('#shortDesc').summernote({
    tabsize: 2,
    minHeight: 200,
    callbacks: {
        onChange: function(contents, $editable) {
            /*if($('#shortDesc').summernote('isEmpty')){
                @this.set('shortDesc', '');                
            } else { 
                @this.set('shortDesc', contents);
            }*/ //// I thought that might cause the page refresh problem, but the problem stays without it as well
            @this.set('shortDesc', contents);
        }
    }
});
```

**Comment by SigalZ** [comment]
OMG!!! My scripts were not wrapped in a function. adding this: $(function() { and page refresh works.
I was not sure if with @push('child-scripts') I still need to add this, it appears I do.
The error still displays only with this check: @if($errors->has('blendDesc')) and not @error('shortDesc')

**Reply by SigalZ** [reply]
And nope, testing a few more times, errors do not display again, page refresh doesn't work. I'm done.

**Comment by SigalZ** [comment]
Ok, sorry for all the posts, got really frustrated here
I think I really made it work now. There were some mistakes in my code but that's the final code if it can help anyone:
The view:
```
<div class="form-group" wire:ignore>
                <label class="control-label" for="shortDesc">Short Desctiption</label>
                <textarea wire:model="shortDesc" class="form-control" name="short_desc" id="shortDesc">{!! $blend->short_desc !!}</textarea>
           </div>
            
            <div role="error" style="font-size: 80%; color: #dc3545" class="mb-4">
                @if($errors->has('shortDesc'))    
                    <strong>{{ $errors->first('shortDesc') }}</strong>
                @endif
            </div>
        
            <div class="form-group" wire:ignore>
                <label class="control-label" for="blendDesc">Blend Description</label>
                <textarea class="form-control" wire:model="blendDesc" id="blendDesc">{!! $blend->coffee_desc !!}</textarea>
            </div>
            <div role="error" style="font-size: 80%; color: #dc3545" class="mb-4">
                @if($errors->has('blendDesc'))    
                    <strong>{{ $errors->first('blendDesc') }}</strong>
                @endif
            </div>
```
The javascript:
```
$(function() {
    $('#shortDesc').summernote({
        tabsize: 2,
        minHeight: 200,
        callbacks: {
            onChange: function(contents, $editable) { 
                if($('#shortDesc').summernote('isEmpty')){
                    @this.set('shortDesc', '');
                } else { 
                    @this.set('shortDesc', contents);
                }
            }
        }
    });
    
    $('#blendDesc').summernote({
        tabsize: 2,
        minHeight: 200,
        callbacks: {
            onChange: function(contents, $editable) {
                if($('#blendDesc').summernote('isEmpty')){
                    @this.set('blendDesc', '');
                } else { 
                    @this.set('blendDesc', contents);
                }
            }
        }    
    });
    $('.dropdown-toggle').dropdown();// For summernote editor buttons to work properly
});
```
I hope I am not mistaken and find it doesn't work again in 5 minutes.... 
God bless us all, pffffff

**Reply by joshhanley** [reply]
@SigalZ all good! Yeah getting other javascript libraries working with Livewire can be a challenge at times. That's why Alpine was written to try and help with some of that crossover but it can still be tough.

Thanks for sharing your solution!
